inc/commonitilobject.class.php

```diff
@@ -8137,6 +8149,22 @@ public function getAssociatedDocumentsCriteria($bypass_rights = false): array {
          ];
       }
 
+      // documents associated to ticketvalidation
+      if (TicketValidation::canView() && $this->getType() == Ticket::getType()) {
+         $or_crits[] = [
+            Document_Item::getTableField('itemtype') => TicketValidation::getType(),
+            Document_Item::getTableField('items_id') => new QuerySubQuery(
+               [
+                  'SELECT' => 'id',
+                  'FROM'   => TicketValidation::getTable(),
+                  'WHERE'  => [
+                     TicketValidation::getTableField('tickets_id') => $this->getID(),
+                  ],
+               ]
+            ),
+         ];
+      }
```

Comment:
It should be handled in a generic way.

```suggestion
      // documents associated to CommonITILValidation
      $validation_class = static::getType() . 'Validation';
      if ($validation_class::canView()) {
         $or_crits[] = [
            Document_Item::getTableField('itemtype') => $validation_class::getType(),
            Document_Item::getTableField('items_id') => new QuerySubQuery(
               [
                  'SELECT' => 'id',
                  'FROM'   => $validation_class::getTable(),
                  'WHERE'  => [
                     $validation_class::getTableField(static::getForeignKeyField()) => $this->getID(),
                  ],
               ]
            ),
         ];
      }
```

Reply:
done